[BugFix] Restore primary key table cause meta inconsistent when restart BE(#30129)

srlch opened the pull request:
Problem:
This problem is introduced by commit cfba8973f4718c0a4f2d6846d2bdaa473ba0be49 In this pr, we create a new tablet meta with snapshot tablet_schema to avoid the crash if the backup table has been done schema change.

But the problem is that, the _updates in new tablet meta is nullptr. It forget to reset it to tablet->updates(). So when remove expired versions for primary key table, tablet_meta.save_meta() will save the wrong meta data for this tablet. When we restart be, it will cause the meta inconsistent problem.

Solution:
Call release_updates to reset the _updates in tablet_meta to tablet->updates()

Fixes #30129

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

trueeyu commented on be/src/runtime/snapshot_loader.cpp:
```diff
@@ -558,6 +558,8 @@ Status SnapshotLoader::primary_key_move(const std::string& snapshot_path, const
     new_tablet_meta->to_meta_pb(&metapb);
     new_tablet_meta->init_from_pb(&metapb, &snapshot_meta.tablet_meta().schema());
+    // make sure _upadtes in tablet_meta is reset to _updates in tablet
+    new_tablet_meta->release_updates(tablet->updates());
```
May be mem leak?

wanpengfei-git commented:
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/snapshot_loader.cpp|0|1|00.00%|[562]|

wanpengfei-git commented:
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/snapshot_loader.cpp|0|1|00.00%|[564]|

wanpengfei-git commented:
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/snapshot_loader.cpp|0|1|00.00%|[564]|

srlch committed:
[BugFix] Restore primary key table cause meta inconsistent when restart BE(#30129)

Problem:
This problem is introduced by commit cfba8973f4718c0a4f2d6846d2bdaa473ba0be49
In this pr, we create a new tablet meta with snapshot tablet_schema to avoid the
crash if the backup table has been done schema change.

But the problem is that, the _updates in new tablet meta is nullptr. It forget to
reset it to tablet->updates(). So when remove expired versions for primary key table,
tablet_meta.save_meta() will save the wrong meta data for this tablet. When we restart
be, it will cause the meta inconsistent problem.

Solution:
reset the tablet_schema directly for the tablet_meta

Signed-off-by: srlch <linzichao@starrocks.com>

wanpengfei-git commented:
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 10 (30.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/snapshot_loader.cpp|0|1|00.00%|[561]|
|:large_blue_circle:|src/storage/tablet_meta.cpp|3|9|33.33%|[256, 542, 543, 544, 546, 548]|

wanpengfei-git commented:
@Mergifyio backport branch-3.1

wanpengfei-git commented:
@Mergifyio backport branch-3.0
